### ISSUE TITLE
yahxemu sez

### ISSUE THREAD

**Issue by new786786** [gheyret/uyghurasr_python#2]
kandak ahwaligez yahxee mu? maxiklandurux kode belan taminlarsezmu? rahmat islirigez barkatlik bolsun

**Comment by new786786** [comment]
test1.wav-->dUxinicinini vaxci qiri vaxupUUxkOpkxk be kUlgUzpahaltijAm yibAkcid vidi
test2.wav-->vabu vOy vAslidA botinnniN GoJxiyA bAximiybi vuGuzniN vowu deyixblAn beJAnnedi
test3.wav-->deyaNb gvaniNli wagigi nigacexeyiNwu giJjijiniNviyyAndey kenayd bi
test4.wav-->pUxizinidi duy bohilog bosi weyjinbodbi
test5.wav-->buniNkUngUdabar beygegA viJow vinpu sAnjinHaniydib ti
test6.wav-->vU daxawuN vi nijun basu veyikiy xinjiyaa be
==================================================
Training Epoch[619]:
[LR:  0.0000200 Loss:  0.04081, Avg loss:  0.03611]: 100%|█████████████████████████████████████████████████████████████████████████| 141/141 [01:14<00:00,  1.90it/s]
test1.wav-->vixiniciniNn baxci qini vayxip xkikxik be kirgizpahaltijAm yibAkjib bdidb
test2.wav-->bibu vOy vAslidAbotininniN GojxiyAbAxJemibi vuGuzniN vuwu beyixmlAn biJAnedi
test3.wav-->bexyaNu gvaneNli nAyeni nigacexinNw giJjijiniNliyAyAnneyjen vi
test4.wav-->pUxinnixlidi bi bohilogboNxi wAyjibebabm
test5.wav-->buNniN kUngUAgcwabegegyAgiJyow vintixAnjinHaniydan bim
test6.wav-->vU duxawuN venijn basiGa veyjiJingjiyaaa be
==================================================
Training Epoch[620]:
[LR:  0.0000200 Loss:  0.03266, Avg loss:  0.03494]:   8%|█████▊                                                                    | 11/141 [00:09<01:05,  2.00it/s]

**Comment by new786786** [comment]
self.uyghur_latin = "AGHJNOUabcdefghijklmnopqrstuvwxyz "

**Comment by new786786** [comment]
import torch
from UModel import UModel
from data import featurelen

MODEL_PATH = "results/UModel_last.pth"
ONNX_PATH = "onnx2.onnx"
device = torch.device("cuda")

model = UModel(num_features_input=featurelen)
checkpoint = torch.load(MODEL_PATH, map_location=device)
model.load_state_dict(checkpoint['st_dict'])
model.to(device)
model.eval()

# 模型真正的输入：梅尔特征 + 长度
BATCH = 1
FEATURE = featurelen
SEQ_LEN = 1000

dummy_mel = torch.randn(BATCH, FEATURE, SEQ_LEN).to(device)
dummy_len = torch.tensor([SEQ_LEN], dtype=torch.int64).to(device)

with torch.no_grad():
    torch.onnx.export(
        model,
        (dummy_mel, dummy_len),
        ONNX_PATH,
        opset_version=16,
        input_names=["mel_input", "input_lengths"],
        output_names=["output"],
        dynamic_axes={
            "mel_input": {0: "batch", 2: "sequence"},
            "input_lengths": {0: "batch"},
            "output": {0: "batch", 1: "sequence"}
        }
    )

print("🎉 ONNX 导出成功！")

**Comment by new786786** [comment]
unimi yahxe qekmide

**Comment by new786786** [comment]
import re
class UycodeConvert:
 def UY2UYL(self, str):
        str = str.replace("ا", "a")
        str = str.replace("ە", "A")
        str = str.replace("ې", "e")
        str = str.replace("ى", "i")
        str = str.replace("و", "o")
        str = str.replace("ۆ", "O")
        str = str.replace("ۇ", "u")
        str = str.replace("ۈ", "U")
        str = str.replace("ئ", "v")
        str = str.replace("ۋ", "w")
        str = str.replace("ر", "r")
        str = str.replace("ي", "y")
        str = str.replace("ڭ", "N")
        str = str.replace("ھ", "h")
        str = str.replace("د", "d")
        str = str.replace("ژ", "J")
        str = str.replace("گ", "g")
        str = str.replace("ج", "j")
        str = str.replace("ل", "l")
        str = str.replace("ز", "z")
        str = str.replace("غ", "G")
        str = str.replace("ب", "b")
        str = str.replace("ن", "n")
        str = str.replace("م", "m")
        str = str.replace("چ", "c")
        str = str.replace("ت", "t")
        str = str.replace("پ", "p")
        str = str.replace("س", "s")
        str = str.replace("ف", "f")
        str = str.replace("خ", "H")
        str = str.replace("ق", "q")
        str = str.replace("ك", "k")
        str = str.replace("ش", "x")
        return str
 def UYL2UY(self, str):
        str = str.replace("a", "ا")
        str = str.replace("A", "ە")
        str = str.replace("e", "ې")
        str = str.replace("i", "ى")
        str = str.replace("o", "و")
        str = str.replace("O", "ۆ")
        str = str.replace("u", "ۇ")
        str = str.replace("U", "ۈ")
        str = str.replace("v", "ئ")
        str = str.replace("w", "ۋ")
        str = str.replace("r", "ر")
        str = str.replace("y", "ي")
        str = str.replace("N", "ڭ")
        str = str.replace("h", "ھ")
        str = str.replace("d", "د")
        str = str.replace("J", "ژ")
        str = str.replace("g", "گ")
        str = str.replace("j", "ج")
        str = str.replace("l", "ل")
        str = str.replace("z", "ز")
        str = str.replace("G", "غ")
        str = str.replace("b", "ب")
        str = str.replace("n", "ن")
        str = str.replace("m", "م")
        str = str.replace("c", "چ")
        str = str.replace("t", "ت")
        str = str.replace("p", "پ")
        str = str.replace("s", "س")
        str = str.replace("f", "ف")
        str = str.replace("H", "خ")
        str = str.replace("q", "ق")
        str = str.replace("k", "ك")
        str = str.replace("x", "ش")
        return str

 def UY2UYL_with_punct(self, str):
        
        punct_map = {
               '؟': '?',
               '！': '!',
               '，': ',',
               '。': '.',
               '؛': ';',
               '：': ':',
        }
        for uy_punct, latin_punct in punct_map.items():
               str = str.replace(uy_punct, latin_punct)

        return self.UY2UYL(str)

**Comment by new786786** [comment]
waktigez yatsa korup bakasez mu
